### PR TITLE
MM-20403 Migrate mfa/setup to typescript

### PR DESCRIPTION
**Pull request by kashifsoofi**
Signed-off-by: Kashif Soofi <kashif.soofi++git@gmail.com>

#### Summary
Migrate all files in `components/mfa/setup` (not sub-directories included) and associated test files to typescript.

#### Ticket Link
GitHub: Fixes https://github.com/mattermost/mattermost-server/issues/16705
JIRA: https://mattermost.atlassian.net/browse/MM-20403

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

### CODE REVIEW

**Review by kashifsoofi — COMMENTED** [review]
index.ts Line 43. getting following error, could not work out the fix, would appreciate any guidance.
```
Argument of type 'typeof Setup' is not assignable to parameter of type 'ComponentType<Matching<{ currentUser: UserProfile; siteName: string | undefined; enforceMultifactorAuthentication: boolean; } & { actions: Actions; }, ClassAttributes<Setup> & Props>>'.
```

File: components/mfa/setup/setup.tsx [hunk 1]
```diff
@@ -3,34 +3,59 @@
...
+                server_error_id: string;
+                message: string;
+            }
+        }>;
+        generateMfaSecret: () => Promise<{
+            data: {
+                secret: string;
+                qr_code: string;
+            };
+            error: {
```
Comment (devinbinnie) [inline components/mfa/setup/setup.tsx:30]: @kashifsoofi Looking at the error message, it looks like the type you have defined here doesn't match the type provided by the redux connector. In the redux connector, the type on `error` is optional, whereas here it is not.

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @kashifsoofi!

**Review by nevyangelova — APPROVED** [review]
LGTM!

**Comment by devinbinnie** [comment]
@furqanmlk Reminder to review this PR :)

**Review by furqanmlk — APPROVED** [review]
All test cases passed ✅ 

image

image

image

**Pull request merged**